### PR TITLE
fix missing legend entries

### LINKED ISSUE #408: ACID(35) xAxisReversed: legend is missing
On ACID(41) xAxisReversed, the legend is missing (R2014a):
image

I was unable to locate an open issue.

Generated code:
```
\begin{tikzpicture}

\begin{axis}[%
width=0.950920245398773\figurewidth,
height=0.895390070921986\figurewidth,
at={(0\figurewidth,0\figurewidth)},
scale only axis,
x dir=reverse,
xmin=0,
xmax=1,
y dir=reverse,
ymin=1,
ymax=2.8,
legend style={at={(0.03,0.03)},anchor=south west,legend cell align=left}
]
\addplot [color=blue,solid]
  table[row sep=crcr]{%
0	1\\
0.01	1.01005016708417\\
...
1	2.71828182845905\\
};
\end{axis}
\end{tikzpicture}%
```

### LINKED ISSUE #648: stackedBarsWithOther (ACID(94)) is missing legend labels
on HG1 only; HG2 and octave are not affected
tested on 7a6b248655f555c1411e7397e47bfa3e50eb8050
image

**edit:** Legend is also missing in test 35 `xAxisReversed` and test 42 `bodeplots`.

### PR DESCRIPTION
**Pull request by PeterPablo**
fixes #408 and #648 
affected tests on HG1:
`bodeplots`, `stackedBarsWithOther` and `xAxisReversed`

image

Ready for review

### CODE REVIEW

**Comment by PeterPablo** [comment]
Travis runs fine, when the other pending PRs are applied (see #666 and https://travis-ci.org/matlab2tikz/matlab2tikz/builds/63005084).

File: test/suites/ACID.Octave.3.8.0.md5 [hunk 1]
```diff
@@ -58,7 +58,7 @@ scatterPlotRandom : 6fadda0940c7c468a1aee0aef8203c4e
 sine_with_annotation : 966fbd52d16fb908384adb9ad3f8d98b
 sine_with_markers : 33e7e752de0d54a80fd95a70fa17a1ef
 spherePlot : 4f8881f1d81962a213aa0526566f976b
-stackedBarsWithOther : 766e986fb790d41c575cb10308025a37
+stackedBarsWithOther : 9fd47e847290b1d5ec435c1279e6e978
```
Comment (egeerardyn) [inline test/suites/ACID.Octave.3.8.0.md5]: Do you have an idea why the Octave hash has changed, while the code change is MATLAB-only (i.e. in `findLegendInfoMATLAB`)? Or is this a pre-existing mismatch?

**Comment by PeterPablo** [comment]
This PR reveals a regression of #641 that influences the output of e.g. `xAxisReversed`:
image

@egeerardyn Should I fix this in this PR or separetly?

**Comment by PeterPablo** [comment]
The brute force approach to resolve this looks like this:
```diff
diff --git a/test/private/initializeGlobalState.m b/test/private/initializeGlobalState.m
index f84ccc2..382f739 100644
--- a/test/private/initializeGlobalState.m
+++ b/test/private/initializeGlobalState.m
@@ -47,6 +47,33 @@ function [orig,cwd] = initializeGlobalState()
     new.ScreenPixelsPerInch.val     = 96;
     new.ScreenPixelsPerInch.ignore  = strcmpi(getEnvironment,'octave');
 
+    % MATLAB's factory values differ from their default values of a clean
+    % MATLAB installation (observed on R2014a, Linux)
+    new.defaultAxesColor.val            = [1 1 1];
+    new.defaultAxesColor.ignore         = false;
+    new.defaultLineColor.val            = [0 0 0];
+    new.defaultLineColor.ignore         = false;
+    new.defaultTextColor.val            = [0 0 0];
+    new.defaultTextColor.ignore         = false;
+    new.defaultAxesXColor.val           = [0 0 0];
+    new.defaultAxesXColor.ignore        = false;
+    new.defaultAxesYColor.val           = [0 0 0];
+    new.defaultAxesYColor.ignore        = false;
+    new.defaultAxesZColor.val           = [0 0 0];
+    new.defaultAxesZColor.ignore        = false;
+    new.defaultFigureColor.val          = [0.8 0.8 0.8];
+    new.defaultFigureColor.ignore       = false;
+    new.defaultPatchEdgeColor.val       = [0 0 0];
+    new.defaultPatchEdgeColor.ignore    = false;
+    new.defaultPatchFaceColor.val       = [0 0 0];
+    new.defaultPatchFaceColor.ignore    = false;
+    new.defaultFigurePaperType.val      = 'A4';
+    new.defaultFigurePaperType.ignore   = false;
+    new.defaultFigurePaperSize.val      = get(0,'defaultFigurePaperSize'); % float value
+    new.defaultFigurePaperSize.ignore   = false;
+    new.defaultFigurePaperUnits.val     = 'centimeters';
+    new.defaultFigurePaperUnits.ignore  = false;
+
     %--- Extract relevant properties and set desired state
     f = fieldnames(new);    % fields of new state
     for i = 1:length(f)
```

* does not change hashes of octave
* affects all hashes of HG1
* affects all hashes of HG2
I'll come up with a PR. This needs to be taken care of, before continuing here.

**Comment by PeterPablo** [comment]
Merged `development` and consequently the above mentioned issue is fixed. I regenerated the hashes and the output looks good to me.
@egeerardyn, please review :-)

**Comment by egeerardyn** [comment]
Works fine on R2014a/b and Octave 3.8 on my Mac.
I also reran the graphical suite on R2014a, and there is only a weird thing with the bode plots, but that's on the reference, not our output. Also, the new global colors are ugly, but it's better than unreliable tests.

screen shot 2015-07-13 at 17 09 17
screen shot 2015-07-13 at 17 09 37
screen shot 2015-07-13 at 17 10 49

**Comment by PeterPablo** [comment]
Wow, those colors are indeed super ugly! I re-merged `develop` and ran `testHeadless` locally to verify things are still fine -- they are all ok.

**Comment by PeterPablo** [comment]
Ah! I think the ugly colors stem from yet another `get(0,'factory...')` annoyance :-/ I did not specify the default colors and colormap. Doh!
If you want to, I can define the default HG2 colors (I had that in the original PR once...).

**Comment by egeerardyn** [comment]
I agree on the probably cause, i.e. some value of `get(0,'Factory...')` or `get(0,'Default...')`. Anyway, that's not the problem here, just something to take note of and correct further on if we find the time.

**Comment by tomlankhorst** [comment]
This seems not to work in my case, legend entries still missing: 

```latex
bode(tf([1],[1 1]));
legend('G0');
cleanfigure
matlab2tikz('legendbode.tex', 'standalone', true)
```

`pdflatex legendbode.tex`

**Pull request merged**